src/shared/bus-wait-for-jobs.c

```diff
@@ -78,13 +85,38 @@ BusWaitForJobs* bus_wait_for_jobs_free(BusWaitForJobs *d) {
         return mfree(d);
 }
 
+int bus_subscribe_with_flags_cached(sd_bus *bus) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        static int cached = -1;
+        int r;
+
+        assert(bus);
+
+        if (cached >= 0)
+                return cached;
+
+        r = bus_call_method(bus,
+                        bus_systemd_mgr,
+                        "SubscribeWithFlags",
+                        &error, &reply, "t", 0);
+        if (r < 0 && !sd_bus_error_has_name(&error, SD_BUS_ERROR_UNKNOWN_METHOD))
+                return log_error_errno(r, "Failed to call SubscribeWithFlags: %s", bus_error_message(&error, r));
+
+        return cached = r >= 0;
+}
```

Comment:
Hmm, I think we should probably update bus_wait_for_jobs() to issue `Subscribe()` too, and remove the explicit calls from logind/machined. That#d be more correct.